src/main/webapp/app/entities/flair-bi/filter/filter-penal-pin.component.js

```diff
@@ -0,0 +1,177 @@
+                type: $stateParams.id ? 'filters' : 'share-link-filter'
+            },
+                $stateParams.id ? $stateParams.id : $stateParams.viewId
+            );
+        }
+
+
+        function updateHeaderFilter() {
+            var filterParameters;
+            filterParameters = filterParametersService.get();
+            var filter = Object.keys(filterParameters);
+
+            if (filter.length == 0) {
+                vm.dimensions.map(function (item) {
+                    vm.selectedFilter[item.name] = [];
+                })
+            }
+            else {
+                filter.forEach(element => {
+                    vm.dimensions.map(function (item) {
+                        if (item.name === element) {
+                            vm.selectedFilter[element] = filterParameters[element]
+                                .map(function (item) {
+                                    return {
+                                        label: item,
+                                        id: item
+                                    }
+                                });
+                        }
+                    })
+
+                });
+            }
+        }
+
+        function applyFilter() {
+            var filter = Object.keys(vm.selectedFilter);
+            var filterParameters = filterParametersService.getSelectedFilter();
+
+            filter.forEach(element => {
+                if (vm.selectedFilter[element].length > 0) {
+                    var dimension = findDimension(element)[0];
+                    if (!filterParameters[dimension.name]) {
+                        filterParameters[dimension.name] = [];
+                    }
+                    filterParameters[dimension.name] = vm.selectedFilter[element].map(function (item) {
+                        return item.id;
+                    });
+                    filterParameters[dimension.name]._meta = {
+                        dataType: dimension.type,
+                        valueType: 'valueType'
+                    };
+                    filterParametersService.saveSelectedFilter(filterParameters);
+
+                }
+            });
+            FilterStateManagerService.add(angular.copy(filterParametersService.get()));
+            filters();
+        }
+
+        function filters() {
+            filterParametersService.save(filterParametersService.getSelectedFilter());
+            $rootScope.updateWidget = {};
+            $rootScope.$broadcast('flairbiApp:filter');
+            $rootScope.$broadcast('flairbiApp:filter-add');
+            //addFilterInIframeURL();
+            $rootScope.$broadcast("flairbiApp:filterClicked");
+            $rootScope.$broadcast("flairbiApp:add-filter-In-FinterPenal");
+
+        }
+
+        function addFilterInIframeURL() {
+            var filters = filterParametersService.getSelectedFilter();
+            filterParametersService.setFilterInIframeURL(filters, vm.iframes, vm.dimensions);
```

Comment:
no need to pass filters..its already a part of service

Reply:
done